app/forms/questionnaire_form.py

```diff
@@ -93,8 +93,23 @@ def answer_errors(self, input_id):
         setattr(QuestionnaireForm, answer_id, field)
 
     if data:
-        form = QuestionnaireForm(MultiDict(data), meta={'csrf': False})
+        form_data = MultiDict(data)
+        clear_other_text_field(block_json, data, form_data)
+
+        form = QuestionnaireForm(form_data, meta={'csrf': False})
     else:
         form = QuestionnaireForm(meta={'csrf': False})
 
     return form
+
+
+def clear_other_text_field(block_json, data, form_data):
```

Comment:
At the moment, the form_helper has some deserialisation logic around dates within it. I would think that this clearing of the other text field would be better placed alongside as the data wouldn't be manipulated as part of the form logic then.